### PR TITLE
chore(cloudsql): update samples to gen2 functions and v2 proxy

### PR DESCRIPTION
**Pull request by jackwotherspoon**
Updating Cloud SQL samples to use gen2 Cloud Functions over gen1 and to use the v2 Cloud SQL Auth Proxy over v1.

### CODE REVIEW

File: cloud-sql/postgres/sqlalchemy/README.md [hunk 1]
```diff
@@ -1,132 +1,356 @@
...
 
+This demo application can be used to connect to Cloud SQL in two different ways:
+
+1. [The Cloud SQL Python Connector](https://github.com/GoogleCloudPlatform/cloud-sql-python-connector) (recommended)
+
+2. [The Cloud SQL Auth Proxy](https://github.com/GoogleCloudPlatform/cloud-sql-proxy)
+
+Using the Cloud SQL Python Connector library is recommended over the Cloud SQL Auth
+Proxy as it provides all the same functionality and features but as a native
+Python package, [cloud-sql-python-connector package](https://pypi.org/project/cloud-sql-python-connector/).
```
Comment (enocom) [inline cloud-sql/postgres/sqlalchemy/README.md:11]: Nit: End with a period. And then, "See cloud-sql-python-connector package."

File: cloud-sql/postgres/sqlalchemy/README.md [hunk 2]
```diff
@@ -1,132 +1,356 @@
...
+To run the demo application locally using the Cloud SQL Python Connector, set
+environment variables and install dependencies as shown below.
+
+Note: The `INSTANCE_CONNECTION_NAME` for your instance can be found on the
+**Overview** page for your instance in the
+[Google Cloud console](https://console.cloud.google.com/sql) or by running
+the following command:
+
+```sh
+gcloud sql instances describe <INSTANCE_NAME>
```
Comment (enocom) [inline cloud-sql/postgres/sqlalchemy/README.md:43]:
```
gcloud sql instances describe <INSTANCE_NAME> --format='value(connectionName)'
```

is better

File: cloud-sql/postgres/sqlalchemy/README.md [hunk 3]
```diff
@@ -1,132 +1,356 @@
...
+
 Note: Saving credentials in environment variables is convenient, but not secure - consider a more
 secure solution such as [Secret Manager](https://cloud.google.com/secret-manager/docs/overview) to
 help keep secrets safe.
 
 Then use this command to launch the proxy in the background:
+
 ```bash
-./cloud_sql_proxy -instances=<PROJECT-ID>:<INSTANCE-REGION>:<INSTANCE-NAME>=tcp:5432 -credential_file=$GOOGLE_APPLICATION_CREDENTIALS &
+./cloudsql-proxy <PROJECT_ID>:<INSTANCE_REGION>:<INSTANCE_NAME> &
```
Comment (enocom) [inline cloud-sql/postgres/sqlalchemy/README.md:263]: cloud-sql-proxy

File: cloud-sql/postgres/sqlalchemy/README.md [hunk 4]
```diff
@@ -1,132 +1,356 @@
...
+
 Note: Saving credentials in environment variables is convenient, but not secure - consider a more
 secure solution such as [Secret Manager](https://cloud.google.com/secret-manager/docs/overview) to
 help keep secrets safe.
 
 Then use this command to launch the proxy in the background:
+
 ```bash
-./cloud_sql_proxy -dir=./cloudsql --instances=<PROJECT-ID>:<INSTANCE-REGION>:<INSTANCE-NAME> --credential_file=$GOOGLE_APPLICATION_CREDENTIALS &
+./cloud-sql-proxy --unix-socket=./cloudsql <PROJECT_ID>:<INSTANCE_REGION>:<INSTANCE_NAME> &
```
Comment (enocom) [inline cloud-sql/postgres/sqlalchemy/README.md:319]: --unix-socket /cloudsql

**Review by enocom — CHANGES_REQUESTED** [review]

**Review by enocom — APPROVED** [review]

**Review by engelke — APPROVED** [review]

**Pull request merged**